payment/p2tr.go

```diff
@@ -0,0 +1,170 @@
+type TaprootPaymentData struct {
+	XOnlyTweakedKey    []byte
+	XOnlyInternalKey   []byte
+	RootScriptTreeHash *chainhash.Hash
+	ScriptTree         *taproot.IndexedElementsTapScriptTree
+}
+
+// FromTweakedKey creates a P2TR payment from a tweaked output key
+func FromTweakedKey(
+	tweakedKey *btcec.PublicKey,
+	net *network.Network,
+	blindingKey *btcec.PublicKey,
+) (*Payment, error) {
+	if tweakedKey == nil {
+		return nil, errors.New("tweaked key can't be empty or nil")
+	}
+
+	if net == nil {
+		net = &network.Liquid
+	}
+
+	return &Payment{
+		Network:     net,
+		BlindingKey: blindingKey,
+		Taproot: &TaprootPaymentData{
+			XOnlyTweakedKey: schnorr.SerializePubKey(tweakedKey),
+		},
+	}, nil
+}
+
+func FromTaprootScriptTreeHash(
+	internalKey *btcec.PublicKey,
+	rootHash *chainhash.Hash,
+	net *network.Network,
+	blindingKey *btcec.PublicKey,
+) (*Payment, error) {
+	if internalKey == nil {
+		return nil, errors.New("internal key can't be empty or nil")
+	}
+
+	if net == nil {
+		net = &network.Liquid
+	}
+
+	return &Payment{
+		Network:     net,
+		BlindingKey: blindingKey,
+		Taproot: &TaprootPaymentData{
+			XOnlyInternalKey:   schnorr.SerializePubKey(internalKey),
+			RootScriptTreeHash: rootHash,
+		},
+	}, nil
+}
+
+func FromTaprootScriptTree(
+	internalKey *btcec.PublicKey,
+	tree *taproot.IndexedElementsTapScriptTree,
+	net *network.Network,
+	blindingKey *btcec.PublicKey,
+) (*Payment, error) {
+	if internalKey == nil {
+		return nil, errors.New("internal key can't be empty or nil")
+	}
+
+	if net == nil {
+		net = &network.Liquid
+	}
+
+	return &Payment{
+		Network:     net,
+		BlindingKey: blindingKey,
+		Taproot: &TaprootPaymentData{
+			XOnlyInternalKey: schnorr.SerializePubKey(internalKey),
+			ScriptTree:       tree,
+		},
+	}, nil
+}
+
+func (p *Payment) taprootBech32() (*address.Bech32, error) {
+	payload := &address.Bech32{
+		Prefix:  p.Network.Bech32,
+		Version: segwitVersion,
+	}
+	if p.Taproot.XOnlyTweakedKey != nil {
+		payload.Program = p.Taproot.XOnlyTweakedKey
+	} else if p.Taproot.XOnlyInternalKey != nil {
+		internalKey, err := schnorr.ParsePubKey(p.Taproot.XOnlyInternalKey)
+		if err != nil {
+			return nil, err
+		}
+
+		if p.Taproot.RootScriptTreeHash != nil {
+			payload.Program = schnorr.SerializePubKey(taproot.ComputeTaprootOutputKey(internalKey, p.Taproot.RootScriptTreeHash.CloneBytes()))
+		} else {
+			if p.Taproot.ScriptTree == nil {
+				payload.Program = schnorr.SerializePubKey(taproot.ComputeTaprootKeyNoScript(internalKey))
+			} else {
+				scriptTreeHash := p.Taproot.ScriptTree.RootNode.TapHash()
+				payload.Program = schnorr.SerializePubKey(taproot.ComputeTaprootOutputKey(internalKey, scriptTreeHash.CloneBytes()))
+			}
+		}
+	}
+
+	if payload.Program == nil {
+		return nil, errors.New("unable to compute taproot's tweaked key from payment data")
+	}
+
+	if len(payload.Program) != 32 {
+		return nil, errors.New("taproot's tweaked key has wrong length")
+	}
```

Comment:
Do we really need these post checks? In general, this lib's methods make sure that everything's ok before doing any operation "by design", instead of performing post checks. 

Is there a way to make these checks over the taproot data before doing the magic in order to make sure that at this point `payload.Program` is exactly a 32-bytes buffer?